### PR TITLE
Add Shopware Environment blog entry

### PR DESCRIPTION
**Pull request by bcremer**
This blog post is stil WIP. I am not finished with the overall structure but would like to get some feedback.
Ping @dnoegel @tiagojsag

### CODE REVIEW

File: source/blog/_posts/2016-01-26-shopware-environment.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+```
+<VirtualHost *:80>
+    DocumentRoot "/home/bcremer/www/"
+    SetEnv SHOPWARE_ENV dev
+</VirtualHost>
+```
+
+### Nginx
+When using the [shopware-with-nginx](https://github.com/bcremer/shopware-with-nginx) configuration:
```
Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: Shopware with nginx

File: source/blog/_posts/2016-01-26-shopware-environment.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+### Shopware Console
+
+This also works for the [Shopware Console Commands](/developers-guide/shopware-5-cli-commands/):
+
+
+```bash
+$ SHOPWARE_ENV=dev ./bin/console sw:generate:attributes
+```
+
+Alternativly you can also use the `--env` option that exists for every console command:
```
Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: Alternatively

File: source/blog/_posts/2016-01-26-shopware-environment.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+```bash
+$ ./bin/console sw:generate:attributes --env=dev
+```
+
+### Export Environment globally
```
Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: fix casing

File: source/blog/_posts/2016-01-26-shopware-environment.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+`$HOME/.profile`
+
+```
+export SHOPWARE_ENV=dev
+```
+
+## How the enviroment is handled
+
+This environment variable is picked by by [`shopware.php`](https://github.com/shopware/shopware/blob/5.2/shopware.php#L92) and [`bin/console`](https://github.com/shopware/shopware/blob/5.2/bin/console#L34) and then propagated to the kernel:
```
Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: by by

File: source/blog/_posts/2016-01-26-shopware-environment.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+$environment = getenv('SHOPWARE_ENV') ?: 'production';
+$kernel = new Kernel($environment);
+```
+
+As you can see, if no environment is given, `production` is used as the default.
+
+
+## Generated files
+
+The environment taken into account for the cache directories and log files.
```
Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: The environment is taken into

File: source/blog/_posts/2016-01-26-shopware-environment.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+Running Shopware each with `dev`, `production` and `testing` environment will result in the following directory tree:
```
Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: Running Shopware once with each of the environments (`dev`, `production` and `testing`) will...

File: source/blog/_posts/2016-01-26-shopware-environment.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+    ├── core_production-2016-01-27.log
+    ├── core_testing-2016-01-25.log
+    ├── core_testing-2016-01-26.log
+    └── core_testing-2016-01-27.log
+```
+
+## Config loading
+
+The config loading order is defined in [engine/Shopware/Configs/Default.php](https://github.com/shopware/shopware/blob/cbafdd378de2bf5afab7504a90a2bc184ebbd681/engine/Shopware/Configs/Default.php#L3).
+The config loader is looking for a file named `config_[ENVIRONMENT].php` first. If that file does not exists it falls back to `config.php`.
```
Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: is looking => looks

File: source/blog/_posts/2016-01-26-shopware-environment.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+On my development machine all my installation contain a file name `config_dev.php`.
```
Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: name => named

File: source/blog/_posts/2016-01-26-shopware-environment.md (hunk 1 above)
Comment (bcremer) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: I would like to keep that. It should reflect the name of the github repo.

Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: ah ok, didn't understand your original intention with it, and assumed it was just some copy paste gone wrong

File: source/blog/_posts/2016-01-26-shopware-environment.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    ├── core_testing-2016-01-26.log
+    └── core_testing-2016-01-27.log
+```
+
+## Config loading
+
+The config loading order is defined in [engine/Shopware/Configs/Default.php](https://github.com/shopware/shopware/blob/cbafdd378de2bf5afab7504a90a2bc184ebbd681/engine/Shopware/Configs/Default.php#L3).
+The config loader looks for a file named `config_[ENVIRONMENT].php` first. If that file does not exists it falls back to `config.php`.
+
+On my development machine all my installation contain a file named `config_dev.php`.
```
Comment (tiagojsag) [inline source/blog/_posts/2016-01-26-shopware-environment.md]: installation => installations

**Pull request merged**